tools/gulp/docs/generatePage.js

```diff
@@ -1,225 +1,38 @@
-
-  ${analytics()}
-</head>
-<body class="ds-base">
-  <div id="js-root">
-    ${componentRenderer()}
-  </div>
-  <script type="text/javascript">
-    window.page = ${JSON.stringify(page)};
-    window.routes = ${JSON.stringify(routes)};
-  </script>
-  <script src="/${rootPath}public/scripts/index.js"></script>
-</body>
-</html>`;
-
-  const pathObj = docsPath(page.referenceURI);
-  return updateFile(html, pathObj);
-}
-
-function generateMarkupPages(page, rootPath) {
-  return generateMarkupPage(page, null, rootPath).then(() => {
+function generateExamples(page, rootPath) {
+  return generateHtmlExample(page, null, rootPath).then(() => {
     if (page.modifiers) {
       return page.modifiers.map(modifier =>
-        generateMarkupPage(page, modifier, rootPath)
+        generateHtmlExample(page, modifier, rootPath)
       );
     }
   });
 }
 
-/**
- * Creates an HTML page with just the KSS section's markup and no additional UI.
- * This can then be viewed in a browser, or rendered in an iFrame in
- * the documentation.
- * @param {Object} page - This should include at least a "markup" property
- * @param {String} rootPath - Root docs site path
- * @return {Promise}
- */
-function generateMarkupPage(page, modifier, rootPath) {
-  if (rootPath) {
-    rootPath = `${rootPath}/`;
-  }
-
-  let id = page.reference; // ie. components.button
-  if (modifier) id += modifier.name; // ie. components.button.ds-c-button--primary
-
-  const markup = processMarkup(page.markup, modifier);
-  const html = `<!DOCTYPE html>
-<html lang="en">
-<head>
-  <meta charset="utf-8">
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <title>Example: ${page.reference}</title>
-  <link rel="stylesheet" href="/${rootPath}public/styles/example.css" />
-
-  ${analytics()}
-</head>
-<body class="ds-base">
-  ${markup}
-  <script type="text/javascript" src="/${rootPath}public/scripts/example.js"></script>
-</body>
-</html>`;
-
-  const uri = `${rootPath}example/${id}`;
-  const pathObj = docsPath(uri);
-  return updateFile(html, pathObj);
-}
-
-/**
- * To ensure we're not unnecessarily regenerating each page each time the
- * generate-pages task is called, we first compare the checksums.
- * @return {Boolean} Should the file be regenerated?
- */
-function checkCache(html, path) {
-  return fs
-    .readFile(path, 'utf8')
-    .then(data => {
-      const fileHash = crypto
-        .createHash('md5')
-        .update(data)
-        .digest('hex');
-      const htmlHash = crypto
-        .createHash('md5')
-        .update(html)
-        .digest('hex');
-      return fileHash !== htmlHash;
-    })
-    .catch(() => true); // File doesn't exist
-}
-
-function docsPath(uri) {
-  if (uri === 'public') throw Error('Filename can\'t be "public"');
-  const dir = path.resolve(__dirname, `../../../docs/${uri}`);
-
-  return {
-    dir: dir,
-    path: path.resolve(dir, 'index.html')
-  };
-}
-
-function saveToFile(html, pathObj, retry = true) {
```

Comment:
ℹ️ Moved to `savePage.js` so it could be shared between the different generate methods